source/s2n/s2n_tls_channel_handler.c

```diff
@@ -57,6 +90,23 @@ void aws_tls_init_static_state(struct aws_allocator *alloc) {
     setenv("S2N_ENABLE_CLIENT_MODE", "1", 1);
     setenv("S2N_DONT_MLOCK", "1", 1);
     s2n_init();
+
+#if OPENSSL_VERSION_LESS_1_1
+    if (!CRYPTO_get_locking_callback()) {
+        s_libcrypto_allocator = alloc;
+        s_libcrypto_locks = aws_mem_acquire(alloc, sizeof(struct aws_mutex) * CRYPTO_num_locks());
+        assert(s_libcrypto_locks);
```

Comment:
why assert when you can exit() 🕶

Reply:
We need that hard crashing assert methinks

Reply:
aws_fatal()